org.eclipse.scout.rt.server.commons/src/main/java/org/eclipse/scout/rt/server/commons/healthcheck/IHealthChecker.java

```diff
@@ -33,6 +36,57 @@ public interface IHealthChecker {
    *          {@link RunContext} used to run the check in.
    * @return <code>true</code> if check was successful, <code>false</code> otherwise.
    */
-  boolean checkHealth(RunContext context);
+  boolean checkHealth(RunContext context, HealthCheckCategoryId category);
+
+  /**
+   * @param category
+   *          non-null category; if no category is available all checks are considered which are marked as
+   *          {@link #isActive()}
+   * @return <code>true</code> if this <code>IHealthChecker</code> accepts the provided category; <code>false</code> to
+   *         exclude this check for this category
+   */
+  default boolean acceptCategory(HealthCheckCategoryId category) {
+    return true;
+  }
+
+  /**
+   * Identifier for a specific check category.
+   *
+   * @see IHealthCheckCategoryProvider
+   */
+  public static class HealthCheckCategoryId extends AbstractStringId {
+
+    private static final long serialVersionUID = -5213735626992608894L;
+
+    protected HealthCheckCategoryId(String id) {
+      super(id);
+    }
+  }
+
+  /**
+   * Interfaces for providers which define health check categories
+   */
+  @ApplicationScoped
+  public static interface IHealthCheckCategoryProvider {
```

Comment:
Alternative implementation suggestion:

```
  @ApplicationScoped
  public static interface IHealthCheckCategory {
      HealthCheckCategoryId getId();
  }

  // --------------------------------------

  public static final class Startup implements IHealthCheckCategory {
    public static final HealthCheckCategoryId ID = HealthCheckCategoryId.of("startup");

    @Override
    public HealthCheckCategoryId getId() {
      return ID;
    }
  }

  public static final class Readiness implements IHealthCheckCategory {
    public  static final HealthCheckCategoryId ID = HealthCheckCategoryId.of("readiness");;

    @Override
    public HealthCheckCategoryId getId() {
      return ID;
    }
  }

  public static final class Liveness implements IHealthCheckCategory {
    public  static final HealthCheckCategoryId ID =  HealthCheckCategoryId.of("liveness");

    @Override
    public HealthCheckCategoryId getId() {
      return ID;
    }
  }
```

Reply:
Fixed in 184fa26d2119186562032245b73ffa38f35d5b9e